src/rust-crypto/rust-crypto.ts

```diff
@@ -841,7 +841,19 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, CryptoEventH
             await this.secretStorage.store("m.cross_signing.self_signing", crossSigningPrivateKeys.self_signing_key);
         }
 
-        if (setupNewKeyBackup) {
+        // likewise with the key backup key: if we have one, store it in secret storage (if it's not already there)
+        // also don't bother storing it if we're about to set up a new backup
+        if (!setupNewKeyBackup) {
+            const keyBackupKey = await this.getSessionBackupPrivateKey();
+
+            const keyBackupKeyInStorage = await secretStorageCanAccessSecrets(this.secretStorage, [
+                "m.megolm_backup.v1",
+            ]);
+
+            if (keyBackupKey && !keyBackupKeyInStorage) {
+                await this.secretStorage.store("m.megolm_backup.v1", encodeBase64(keyBackupKey));
```

Comment:
There should be an example in when you resetKeyBackup(). I think that is fine, maybe you could get it from the rust olmMachine directly to avoid a base64 -> byte array -> base64?

Shouldn't we check more than `keyBackupKeyInStorage` ? What if the 4S contains an old key it won't be updated? Maybe we should check that the key in cache is valid before adding it to 4S? Is the backup trusted?

Is there a link to an issue for this PR?

Reply:
Okay, I think that should be better. I've made it overwrite the key if different, as per the comment.

This came out as part of the key backup by default work, so it's not a huge bug currently, but would break the flow of setting up 4S after a key backup, hence no issue.